src/Shared/PackageManagers/NPMProjectManager.cs

```diff
@@ -112,20 +112,48 @@ public override async Task<IEnumerable<string>> EnumerateVersions(PackageURL pur
                 HttpClient httpClient = CreateHttpClient();
 
                 JsonDocument doc = await GetJsonCache(httpClient, $"{ENV_NPM_API_ENDPOINT}/{packageName}", useCache);
+                Dictionary<string, DateTime> versionDictionary = new();
+                if(doc.RootElement.TryGetProperty("time", out JsonElement time))
+                {
+                    foreach (JsonProperty versionKey in time.EnumerateObject())
+                    {
+                        if (!versionKey.NameEquals("modified") && !versionKey.NameEquals("created"))
+                        {
+                            Logger.Debug("Identified {0} version {1}, published on {2}.", packageName, versionKey.Name, versionKey.Value.GetString());
+                            versionDictionary.TryAdd(versionKey.Name, DateTime.Parse(versionKey.Value.GetString()));
+                        }
+                    }
+                }
+
                 List<string> versionList = new();
 
                 foreach (JsonProperty versionKey in doc.RootElement.GetProperty("versions").EnumerateObject())
                 {
                     Logger.Debug("Identified {0} version {1}.", packageName, versionKey.Name);
                     versionList.Add(versionKey.Name);
                 }
+
                 string? latestVersion = doc.RootElement.GetProperty("dist-tags").GetProperty("latest").GetString();
                 if (!string.IsNullOrWhiteSpace(latestVersion))
                 {
                     Logger.Debug("Identified {0} version {1}.", packageName, latestVersion);
                     versionList.Add(latestVersion);
                 }
-                return SortVersions(versionList.Distinct());
+
+                foreach ((string version, _) in versionDictionary)
+                {
+                    if (!versionList.Contains(version))
+                    {
+                        versionDictionary.Remove(version);
+                    }
+                }
```

Comment:
I think the simplest thing to do instead of this loop would be:

`versionDictionary = versionDictionary.Where((version, _) => versionList.Contains(key));`